### PR TITLE
Add check-control-plane-machine-name to check-cluster-health chain to make sure masters are not recreated

### PR DESCRIPTION
**Pull request by sunzhaohua2**
Add health check chain to existing workflows @jhou1 @jianlinliu @yunjiang29 please help to take a look if this makes sense, if so I'll update other workflows, thanks!
Discuss in here: https://redhat-internal.slack.com/archives/CF8SMALS1/p1679641767534869

### CODE REVIEW

**Comment by jianlinliu** [comment]
maybe this will conflicts with what @liangxia is doing in https://github.com/openshift/release/pull/37918, if I remember right, prow ci does not allow call the same step more than one time.

**Comment by sunzhaohua2** [comment]
> maybe this will conflicts with what @liangxia is doing in #37918, if I remember right, prow ci does not allow call the same step more than one time.

Yes, thanks, this will conflict with those jobs. In this pr, I created a chain, add all the check steps to this chain, and then add this chain to the provision chain, so that the check will complete after installation, jobs will not need `test  ref: cucushift-installer-check-cluster-health`. The issue is maybe not every check point is suitable for every version. 

If we add the check in jobs, it will be more flexible for different version, different clusters. what we should do now? Update/add in jobs test with `chain: cucushift-installer-check` for 4.13 aws/gcp/azure, 4.12 aws, do you think this is ok?

**Comment by jianlinliu** [comment]
lxia already introduced `cucushift-installer-check-cluster-health` chain in https://github.com/openshift/release/pull/37971/files, if you like to run `cucushift-installer-check-control-plane-machine-name` checking, I guess we can add the step to `cucushift-installer-check-cluster-health` chain now.

File: ci-operator/step-registry/cucushift/installer/check/cluster-health/cucushift-installer-check-cluster-health-chain.yaml [hunk 1]
```diff
@@ -2,5 +2,6 @@ chain:
   as: cucushift-installer-check-cluster-health
   steps:
   - ref: cucushift-installer-check-cluster-health
+  - ref: cucushift-installer-check-control-plane-machine-name
```
Comment (liangxia) [inline ci-operator/step-registry/cucushift/installer/check/cluster-health/cucushift-installer-check-cluster-health-chain.yaml:5]: Will this check works on all profiles, e.g, sno, hypershift, rosa, aro, ...

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/cluster-health/cucushift-installer-check-cluster-health-chain.yaml:5]: In the `cucushift-installer-check-control-plane-machine-name` step, I saw the following pre-check code:
```
oc get machine -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers -owide 1>${machine_output} 2>${err_output}

if grep -ir 'No resources found in openshift-machine-api namespace.' ${err_output}; then
    echo "WARN: No machines in openshift-machine-api namespace, skip checking"
    exit 0
fi
```
So I guess it should work in most of situations.

Comment (liangxia) [reply ci-operator/step-registry/cucushift/installer/check/cluster-health/cucushift-installer-check-cluster-health-chain.yaml:5]: Ack

**Comment by liangxia** [comment]
Perhaps we need to update PR title to reflect the code change, too ?

**Comment by sunzhaohua2** [comment]
> Perhaps we need to update PR title to reflect the code change, too ?

Done

**Comment by liangxia** [comment]
I can see Yunfei used a hold above, just in case he has concerns

/assign @yunjiang29

**Pull request merged**